hw3/hw3-react/src/houses/HouseChart.js

```diff
@@ -0,0 +1,92 @@
+import { Doughnut } from 'react-chartjs-2';
+import {
+  Chart as ChartJS,
+  ArcElement,
+  Tooltip,
+  Legend,
+  Colors,
+} from 'chart.js';
+
+ChartJS.register(ArcElement, Tooltip, Legend, Colors);
+
+function HouseChart() {
+  const cleanName = function validateAndCleanLastName(houseName) {
+    const regex = /House./;
+    let cleanedName = houseName.replace(regex, '');
+
+    if (
+      cleanedName === '' ||
+      cleanedName === 'None' ||
+      cleanedName === 'Unkown'
+    ) {
+      cleanedName = 'Unknown';
+    } else if (
+      cleanedName === 'Targaryan' ||
+      cleanedName === 'Worm' ||
+      cleanedName === 'Naathi' ||
+      cleanedName === 'Naharis'
+    ) {
+      cleanedName = 'Targaryen';
+    } else if (cleanedName === 'Lanister' || cleanedName === 'Qyburn') {
+      cleanedName = 'Lannister';
+    } else if (cleanedName === 'Lorathi') {
+      cleanedName = 'Lorath';
+    } else if (cleanedName === 'Sand' || cleanedName === 'Viper') {
+      cleanedName = 'Martell';
+    }
+
+    return cleanedName;
+  };
+
+  const reducer = function reduceTheCharactersArray(
+    houseNameCounter,
+    { family },
+  ) {
+    const cleanedName = cleanName(family);
+
+    const currentCount = houseNameCounter[cleanedName] ?? 0;
+    return {
+      ...houseNameCounter,
+      [cleanedName]: currentCount + 1,
+    };
+  };
+
+  const createChart = function createChartDataFromSessionStorage() {
+    const storedCharacters = sessionStorage.getItem('characters');
```

Comment:
Although sessionStorage does retrieve the data information, I would consider passing the data that you receive from the API call to the SearchPage component so that this wouldn't be required. It aligns more with how React should function in the sense that you pass data to other components to use.

Reply:
Thanks, all the props feedback was really helpful! I refactored the way data was managed to follow your logic.